var/spack/repos/builtin/packages/sz/package.py

```diff
@@ -97,87 +97,66 @@ def setup_run_environment(self, env):
         if "+hdf5" in self.spec:
             env.prepend_path("HDF5_PLUGIN_PATH", self.prefix.lib64)
 
-    def _test_2d_float(self):
+    def test_2d_float(self):
         """This test performs simple 2D compression/decompression (float)"""
         test_data_dir = self.test_suite.current_test_data_dir
 
-        filename = "testfloat_8_8_128.dat"
-        orifile = test_data_dir.join(filename)
-
-        exe = "sz"
-        reason = "testing 2D compression of {0}".format(exe)
-        options = ["-z", "-f", "-i", orifile, "-M", "REL", "-R", "1E-3", "-2", "8", "1024"]
-
-        self.run_test(
-            exe,
-            options,
-            [],
-            installed=True,
-            purpose=reason,
-            skip_missing=True,
-            work_dir=test_data_dir,
-        )
-
-        filename = "testfloat_8_8_128.dat.sz"
-        decfile = test_data_dir.join(filename)
-
-        reason = "testing 2D decompression of {0}".format(exe)
-        options = ["-x", "-f", "-i", orifile, "-s", decfile, "-2", "8", "1024", "-a"]
-
-        self.run_test(
-            exe,
-            options,
-            [],
-            installed=True,
-            purpose=reason,
-            skip_missing=True,
-            work_dir=test_data_dir,
-        )
-
-    def _test_3d_float(self):
+        with working_dir(test_data_dir):
+            filename = "testfloat_8_8_128.dat"
+            orifile = test_data_dir.join(filename)
+            exe = which("sz")
+            if exe is None:
+                raise SkipTest(f"sz is not installed for version {self.version}")
+
+            with test_part(self, "test_2d_compression_sz", purpose="testing 2D compression of sz"):
```

Comment:
The test part name needs to start with the name of the test method (eg, `test_2d_float_compression`) for proper results reporting.